models/relation.rb

```diff
@@ -43,16 +43,18 @@ class Relation < Base
 
     def self.create(hash)
       # TODO: remove it after the linked_ids is replaced with linked_service_ids in the api definition
-      exclusive = [:linked_service_ids, :linked_ids, :linked_services].freeze
-      raise "#{exclusive} fields are mutually exclusive" if hash.keys.count{ |k| exclusive.include? k } > 1
+      if hash.key?(:linked_service_ids) && hash.key?(:linked_ids) && hash[:linked_service_ids] != hash[:linked_ids]
+        raise 'linked_service_ids and linked_ids should have the same content if both used'
+      end
+
+      # Raise condition required due to the use of lib/helper deep_copy
+      raise 'linked_services should not belong to vrp hash' if hash.key?(:linked_services)
 
       # TODO: remove it after the linked_ids is replaced with linked_service_ids in the api definition
       if hash.key?(:linked_ids)
         hash[:linked_service_ids] = hash[:linked_ids]
       elsif hash.key?(:linked_service_ids)
         hash[:linked_ids] = hash[:linked_service_ids]
-      elsif hash.key?(:linked_services)
-        hash[:linked_ids] = hash[:linked_services].map(&:id)
```

Comment:
This was to make sure that until we finally fix the `linked_ids` vs `linked_vehicle_ids` disperency -- i.e., firmly pass to `linked_service_ids` -- the model would stay inter-operable. What if I just create the relation with `linked_services` -- i.e., with the proper objects without re-generation then we need to set the linked_ids manually, no ?

Reply:
In case we generate a new relation from a hash. Which field is prime ? `:linked_services` ?

Reply:
I don't know, it depends how it is handled by active_hash. For example we do nothing splecial for vehicle.start_point -- what happens if the has has start_point and start_point_id pointing two diffrent objects. I suspect the one that comes later overwrites the first one in `hash.each{ |key, value| .... }` loop